dev/src/write-batch.ts

```diff
@@ -575,12 +575,33 @@ export class WriteBatch implements firestore.WriteBatch {
       api.BatchWriteResponse
     >('batchWrite', request, tag);
 
-    return (response.writeResults || []).map((result, i) => {
+    const writeResults = response.writeResults || [];
+    let latestTimestamp = Timestamp.fromMillis(0);
+    writeResults.forEach(result => {
```

Comment:
Nit: Prefer using `for ... of` loops when possible, as this makes for far easier debugging than inline lambdas.

Reply:
n/a anymore